__mocks__/@etherspot/modular-sdk.js

```diff
@@ -0,0 +1,143 @@
+    let maxFeePerGas = ethers.utils.parseUnits('1', 'gwei');
+    let maxPriorityFeePerGas = ethers.utils.parseUnits('1', 'gwei');
+    let callGasLimit = ethers.BigNumber.from('50000');
+    let signature = '0x004';
+
+    if (paymaster?.url === 'someUrl') {
+      maxFeePerGas = ethers.utils.parseUnits('2', 'gwei');
+      maxPriorityFeePerGas = ethers.utils.parseUnits('3', 'gwei');
+      callGasLimit = ethers.BigNumber.from('75000');
+    }
+
+    if (paymaster?.url === 'someUnstableUrl') {
+      signature = '0x0';
+    }
+
+    let finalGasLimit = ethers.BigNumber.from(callGasLimit);
+
+    if (this.sdkChainId === 420) {
+      throw new Error('Transaction reverted: chain too high');
+    }
+
+    this.userOps.forEach((userOp) => {
+      if (userOp.to === '0xDEADBEEF') {
+        throw new Error('Transaction reverted: invalid address');
+      }
+      finalGasLimit = finalGasLimit.add(callGasLimit);
+      if (userOp.data && userOp.data !== '0x0' && userOp.data !== '0xFFF') {
+        finalGasLimit = finalGasLimit.add(callGasLimit);
+      }
+    });
+
+    return {
+      sender: defaultAccountAddress,
+      nonce: this.nonce,
+      initCode: '0x001',
+      callData: '0x002',
+      callGasLimit: finalGasLimit,
+      verificationGasLimit: ethers.BigNumber.from('25000'),
+      preVerificationGas: ethers.BigNumber.from('75000'),
+      maxFeePerGas,
+      maxPriorityFeePerGas,
+      paymasterAndData: '0x003',
+      signature,
+    };
+  }
+
+  totalGasEstimated({ callGasLimit, verificationGasLimit, preVerificationGas }) {
+    return callGasLimit.add(verificationGasLimit).add(preVerificationGas);
+  }
+
+  async send(userOp) {
+    if (this.sdkChainId === 696969) {
+      throw new Error('Transaction reverted: chain too hot');
+    }
+
+    if (userOp.signature === '0x0') {
+      throw new Error('Transaction reverted: invalid signature');
+    }
+
+    /**
+     * provide fake userOp hash by increasing nonce on each send
+     * and add SDK chain ID to make it more unique per userOp
+     */
+    const userOpHash = this.nonce.add(this.sdkChainId).toHexString();
+    this.nonce = this.nonce.add(1);
+
+    return userOpHash;
+  }
+
+  async installModule(moduleType, module, initData, accountAddress) {
+    if (!accountAddress && !defaultAccountAddress) {
+      return 'No account address provided!'
+    }
+
+    if (!moduleType || !module) {
+      return 'installModule props missing'
+    }
+
+    if (module === '0x222') {
+      return 'module is already installed'
+    }
+
+    return '0x123';
+  }
+
+  async uninstallModule(moduleType, module, deinitData, accountAddress) {
+    if (module === '0x222') {
+      return 'module is not installed'
+     }
+
+    if (!accountAddress && !defaultAccountAddress) {
+      return 'No account address provided!'
+    }
+
+    if (!moduleType || !module || !deinitData) {
+     return 'uninstallModule props missing'
+    }
+
+    return '0x456';
+  }
```

Comment:
Can i just check, should these not be throwing errors as opposed to returning a string? Is this how the Modular SDK behaves?

Reply:
Yes you are right, still experiencing and learning on Jest, I've changed this to throw errors instead and adapted the test accordingly.